### ISSUE TITLE
Determine structure of output files and add process for writing unfiltered and filtered output

### ISSUE THREAD

**Issue by allyhawkins** [AlexsLemonade/alsf-scpca#107]
After each sample has gone through pre-processing, the data will be contained in an output folder containing all information about the alevin-fry run and output matrix files (currently in `.mtx` format). We plan to provide both the unfiltered and filtered output to users, so we will need to decide if we want to provide the original `.mtx `files, `SingleCellExperiment `objects, or both? 

We will also need to decide on the structure of the output folders for the unfiltered and filtered output and what files we want to include. After pre-processing, the next process in the scpca nextflow pipeline should be creating the unfiltered and filtered output in the proper format.

**Comment by jashapiro** [comment]
I had been assuming we would provide `SingleCellExperiment` objects to allow people to read in more fully processed data, but that might not be the most compatible, if we want to support other downstream workflows. 

One possibility to look into is `.loom` files as used by https://www.humancellatlas.org/, which is really just a form of hdf5.
This should be pretty easy to generate from a SingleCellExperiment, and can be read in by Seurat, scanpy, etc.

More info: 
http://loompy.org/
https://bioconductor.org/packages/release/bioc/vignettes/LoomExperiment/inst/doc/LoomExperiment.html

**Comment by allyhawkins** [comment]
In thinking about providing example counts data for AlexsLemonade/ScPCA-admin#182, some more concrete questions have come to mind: 

1. For each sample do we just want to provide data in either a SCE or loom object? or is there rationale for providing the original output from the alevin folder that has not been pulled into R yet as well? I could see how both could be beneficial to someone.
2. In terms of structure of the folder, I was thinking that for each sample we would have a folder that contained a filtered and unfiltered subfolder. Within each of those folders were the files associated with the filtered or unfiltered counts matrix. Is there any reason that we should be combining the two into one R object instead rather than keeping them separate? If we also wanted to include the matrix files, we could have a separate folder for that. 
3. Besides the actual counts files (either SCE/loom and/or matrix files) are there any other files that should be included in each sample's folder that I am forgetting?

**Comment by jashapiro** [comment]
> In thinking about providing example counts data for [AlexsLemonade/ScPCA-admin#182](https://github.com/AlexsLemonade/ScPCA-admin/issues/182), some more concrete questions have come to mind:
> 
> 1. For each sample do we just want to provide data in either a SCE or loom object? or is there rationale for providing the original output from the alevin folder that has not been pulled into R yet as well? I could see how both could be beneficial to someone.

While there is some value to providing the full alevin output, I would tend to think the added complexity will not be worth it for the vast majority of users. If we provide sufficient detail on the process, I would not expect there is too much the rest of the folder will provide. 


> 2. In terms of structure of the folder, I was thinking that for each sample we would have a folder that contained a filtered and unfiltered subfolder. Within each of those folders were the files associated with the filtered or unfiltered counts matrix. Is there any reason that we should be combining the two into one R object instead rather than keeping them separate? If we also wanted to include the matrix files, we could have a separate folder for that.

My hope was that we would be able to provide a single file, either filtered or unfiltered. I do not think there is any value in combining them. While I can imagine somebody downloading both initially to compare, I would expect that most users will quickly know which one they want, and doubling the size of the download doesn't seem like a great idea.

The other advantage of using a loom file (SCE-derived) is that we should be able to provide CITE-seq data within that single file. Otherwise, we would have to provide multiple matrix files in the case of CITE-seq files. 

> 3. Besides the actual counts files (either SCE/loom and/or matrix files) are there any other files that should be included in each sample's folder that I am forgetting?

We should include the QC output file, and any provenance summaries that we decide are required. 

I was hoping we could produce a fairly simple output folder like this:
```
SCPCS0000
├── filtered.loom
├── qc_report.html
└── filtered.loom
```
Where the download would include either `filtered.loom` or `filtered.loom` but not both. (And `.loom `is not final, but could be .RDS or some other format... we still need to explore the `.loom` output a bit more.)

If we were to include matrix files, we would have to do something like this:
```
SCPCS0001
├── filtered
│   ├── adt_count.barcodes
│   ├── adt_count.features
│   ├── adt_count.mtx
│   ├── gene_count.barcodes
│   ├── gene_count.genes
│   └── gene_count.mtx
├── qc_report.html
└── unfiltered
    ├── adt_count.barcodes
    ├── adt_count.features
    ├── adt_count.mtx
    ├── gene_count.barcodes
    ├── gene_count.genes
    └── gene_count.mtx
```
Which isn't terrible, but is a lot more to deal with. If we are providing `.mtx` files we probably will also may end up wanting to provide some of the per gene or per cell stats that we can attach for "free" in a loom or rds file.

**Comment by allyhawkins** [comment]
I'm fine with not including the other files and just having either an rds or loom file. I agree that it combines more complexity. 

> My hope was that we would be able to provide a single file, either filtered or unfiltered. I do not think there is any value in combining them. 

I thought we had decided that we DO want an unfiltered and a filtered output for people to choose from? So in that case the output folder for each sample would look as follows and then people could choose if they wanted to download filtered or unfiltered version. Am I misinterpreting this? 

Thanks for the reminder about the qc report, I knew I was forgetting something! 

```
SCPCS0000
├── filtered.loom
├── qc_report.html
└── unfiltered.loom
```

**Comment by jashapiro** [comment]
> I thought we had decided that we DO want an unfiltered and a filtered output for people to choose from? So in that case the output folder for each sample would look as follows and then people could choose if they wanted to download filtered or unfiltered version. Am I misinterpreting this?

Yes, we want to provide the option. So I guess there is the output file from the pipeline, and then the files that somebody would download. _We_ would provide both, but the website would only serve one at a time.
